my_game.py

```diff
@@ -254,9 +254,16 @@ def on_update(self, delta_time):
         elif self.right_pressed and not self.left_pressed:
             self.player_sprite.angle -= PLAYER_ROTATE_SPEED
 
+        # Player rocket engine
+        if self.up_pressed:
+            # rotate player graphics to match direction
+            self.player_sprite.change_x += self.player_sprite.speed * cos(self.player_sprite.radians + pi/2)
+            self.player_sprite.change_y += self.player_sprite.speed * sin(self.player_sprite.radians + pi/2)
+        
+
         # Move player with joystick if present
         if self.joystick:
-            self.player_sprite.change_x = round(self.joystick.x) * PLAYER_SPEED_X
+            self.player_sprite.change_x = round(self.joystick.x) * PLAYER_SPEED
```

Comment:
This code is wrong. Test with a joypad, and make sure it also works for that.

Reply:
Don't want to :(

I will make a new issue, and get another person to fix my problems :+1: